File: app/src/main/java/fmrsabino/moviesdb/data/model/configuration/Configuration.java
package fmrsabino.moviesdb.data.model.configuration;

import android.os.Parcelable;

import com.google.auto.value.AutoValue;
import com.squareup.moshi.JsonAdapter;
import com.squareup.moshi.Moshi;

import java.util.List;

@AutoValue
public abstract class Configuration implements Parcelable {
    public abstract Image images();

    public static JsonAdapter<Configuration> typeAdapter(Moshi moshi) {
        return new AutoValue_Configuration.MoshiJsonAdapter(moshi);
    }

    public static Builder builder() {
        return new AutoValue_Configuration.Builder();
    }

    @AutoValue.Builder
    public abstract static class Builder {
        public abstract Builder images(Image image);
        public abstract Configuration build();
    }
}

File: app/src/main/java/fmrsabino/moviesdb/data/model/configuration/Configuration.kt
package fmrsabino.moviesdb.data.model.configuration

import com.squareup.moshi.Json

data class Configuration(val images: Image? = null)

data class Image(
        @Json(name = "base_url") val baseUrl: String? = null,
        @Json(name = "secure_base_url") val secureBaseUrl: String? = null,
        @Json(name = "backdrop_sizes") val backdropSizes: List<String>? = null,
        @Json(name = "poster_sizes") val posterSizes: List<String>? = null,
        @Json(name = "logo_sizes") val logoSizes: List<String>? = null)
